calico_rkt/calico_rkt.py

```diff
@@ -1,138 +1,149 @@
-#!/usr/bin/env python
+# Copyright 2015 Metaswitch Networks
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 from __future__ import print_function
 import socket
 import functools
+import logging
 import json
 import os
 import sys
 from subprocess import check_output, CalledProcessError, check_call
-
 from netaddr import IPAddress, IPNetwork, AddrFormatError
+
+from logutils import configure_logger
 from pycalico import netns
 from pycalico.ipam import IPAMClient, SequentialAssignment
 from pycalico.netns import Namespace
-from pycalico.datastore_datatypes import Rules
+from pycalico.datastore_datatypes import Rules, IPPool
 from pycalico.datastore import IF_PREFIX, DatastoreClient
 from pycalico.datastore_errors import PoolNotFound
 
-print_stderr = functools.partial(print, file=sys.stderr)
+logger = logging.getLogger(__name__)
 
-
-input_ = ''.join(sys.stdin.readlines()).replace('\n', '')
-input_json = json.loads(input_)
-
-# Append to existing env, to avoid losing PATH etc.
-# TODO-PAT: This shouldn't be hardcoded
-env = os.environ.copy()
 ETCD_AUTHORITY_ENV = 'ETCD_AUTHORITY'
-env[ETCD_AUTHORITY_ENV] = 'localhost:2379'
 
-# PROFILE_LABEL = 'CALICO_PROFILE'
-# ETCD_PROFILE_PATH = '/calico/'
+env, INPUT_JSON = {}, {}
 
 ORCHESTRATOR_ID = "rkt"
 HOSTNAME = socket.gethostname()
-NETNS_ROOT= '/var/lib/rkt/pods/run'
+NETNS_ROOT = '/var/lib/rkt/pods/run'
 
 def main():
     mode = env['CNI_COMMAND']
 
     if mode == 'init':
-        print_stderr('No initialization work to perform')
+        logger.info('No initialization work to perform')
     elif mode == 'ADD':
-        print_stderr('Executing Calico pod-creation plugin')
-        create(
-            container_id=env['CNI_CONTAINERID']
-            )
+        logger.info('Executing Calico pod-creation plugin')
```

Comment:
You've got a bit of a redundant logging message here, since you log a second message immediately in the `create` and `delete` functions. Though a lot of things can go wrong in python, we can assume that we'll make it to that next logger, so remove either this logging message or the one down there